core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -1201,142 +1081,170 @@ object ZManaged {
   }
 
   /**
-   * A `FinalizerRef` describes the finalizers associated with a scope.
-   *
+   * A finalizer used in a [[ReleaseMap]]. The [[Exit]] value passed to
+   * it is the result of executing [[ZResource#use]] or an arbitrary value
+   * passed into [[ReleaseMap#release]].
    */
-  trait FinalizerRef[R] {
+  type Finalizer = Exit[Any, Any] => UIO[Any]
+  object Finalizer {
+    val noop: Finalizer = _ => UIO.unit
+  }
+
+  final class IfM[R, E](private val b: ZManaged[R, E, Boolean]) extends AnyVal {
```

Comment:
```scala
ifM (expr) {
  consequentM
} elseM {
  alternateM
}
```
😃

Reply:
Lol. Happy to satisfy your DSL desires in a follow up 😂